File: source/clibCore/Models/Entity/NavigatorEntryModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses
Imports Contensive.Core.Controllers
Imports Newtonsoft.Json

Namespace Contensive.Core.Models.Entity
    Public Class NavigatorEntryModel
        Inherits baseModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "Navigator Entries"
        Public Const contentTableName As String = "ccMenuEntries"
        Private Shadows Const contentDataSource As String = "default"
        '
        '====================================================================================================
        ' -- instance properties
        Public Property ParentID As Integer
        Public Property NavIconTitle As String
        Public Property NavIconType As Integer
        Public Property AddonID As Integer
        Public Property AdminOnly As Boolean
        Public Property ContentID As Integer
        Public Property DeveloperOnly As Boolean
        Public Property EditArchive As Boolean
        Public Property EditBlank As Boolean
        Public Property EditSourceID As Integer
        Public Property HelpAddonID As Integer
        Public Property HelpCollectionID As Integer
        Public Property InstalledByCollectionID As Integer
        Public Property LinkPage As String
        Public Property NewWindow As Boolean
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass) As NavigatorEntryModel
            Return add(Of NavigatorEntryModel)(cpCore)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass, ByRef callersCacheNameList As List(Of String)) As NavigatorEntryModel
            Return add(Of NavigatorEntryModel)(cpCore, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer) As NavigatorEntryModel
            Return create(Of NavigatorEntryModel)(cpCore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer, ByRef callersCacheNameList As List(Of String)) As NavigatorEntryModel
            Return create(Of NavigatorEntryModel)(cpCore, recordId, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String) As NavigatorEntryModel
            Return create(Of NavigatorEntryModel)(cpCore, recordGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String, ByRef callersCacheNameList As List(Of String)) As NavigatorEntryModel
            Return create(Of NavigatorEntryModel)(cpCore, recordGuid, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String) As NavigatorEntryModel
            Return createByName(Of NavigatorEntryModel)(cpCore, recordName)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String, ByRef callersCacheNameList As List(Of String)) As NavigatorEntryModel
            Return createByName(Of NavigatorEntryModel)(cpCore, recordName, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub save(cpCore As coreClass)
            MyBase.save(cpCore)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, recordId As Integer)
            delete(Of NavigatorEntryModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, ccGuid As String)
            delete(Of NavigatorEntryModel)(cpCore, ccGuid)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String, callersCacheNameList As List(Of String)) As List(Of NavigatorEntryModel)
            Return createList(Of NavigatorEntryModel)(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String) As List(Of NavigatorEntryModel)
            Return createList(Of NavigatorEntryModel)(cpCore, sqlCriteria, sqlOrderBy)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String) As List(Of NavigatorEntryModel)
            Return createList(Of NavigatorEntryModel)(cpCore, sqlCriteria)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub invalidatePrimaryCache(cpCore As coreClass, recordId As Integer)
            invalidateCacheSingleRecord(Of NavigatorEntryModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, recordId As Integer) As String
            Return baseModel.getRecordName(Of NavigatorEntryModel)(cpcore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, ccGuid As String) As String
            Return baseModel.getRecordName(Of NavigatorEntryModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordId(cpcore As coreClass, ccGuid As String) As Integer
            Return baseModel.getRecordId(Of NavigatorEntryModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createDefault(cpcore As coreClass) As NavigatorEntryModel
            Return createDefault(Of NavigatorEntryModel)(cpcore)
        End Function
    End Class
End Namespace

File: source/core/Models/DbModels/NavigatorEntryModel.cs
﻿
using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Processor;
using Contensive.Processor.Models.DbModels;
using Contensive.Processor.Controllers;
using static Contensive.Processor.Controllers.genericController;
using static Contensive.Processor.constants;
//
namespace Contensive.Processor.Models.DbModels {
    public class NavigatorEntryModel : baseModel {
        //
        //====================================================================================================
        //-- const
        public const string contentName = "Navigator Entries";
        public const string contentTableName = "ccMenuEntries";
        private const string contentDataSource = "default";
        //
        //====================================================================================================
        // -- instance properties
        public int ParentID { get; set; }
        public string NavIconTitle { get; set; }
        public int NavIconType { get; set; }
        public int AddonID { get; set; }
        public bool AdminOnly { get; set; }
        public int ContentID { get; set; }
        public bool DeveloperOnly { get; set; }



        public int HelpAddonID { get; set; }
        public int HelpCollectionID { get; set; }
        public int InstalledByCollectionID { get; set; }
        public string LinkPage { get; set; }
        public bool NewWindow { get; set; }
        //
        //====================================================================================================
        public static NavigatorEntryModel add(CoreController core) {
            return add<NavigatorEntryModel>(core);
        }
        //
        //====================================================================================================
        public static NavigatorEntryModel add(CoreController core, ref List<string> callersCacheNameList) {
            return add<NavigatorEntryModel>(core, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static NavigatorEntryModel create(CoreController core, int recordId) {
            return create<NavigatorEntryModel>(core, recordId);
        }
        //
        //====================================================================================================
        public static NavigatorEntryModel create(CoreController core, int recordId, ref List<string> callersCacheNameList) {
            return create<NavigatorEntryModel>(core, recordId, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static NavigatorEntryModel create(CoreController core, string recordGuid) {
            return create<NavigatorEntryModel>(core, recordGuid);
        }
        //
        //====================================================================================================
        public static NavigatorEntryModel create(CoreController core, string recordGuid, ref List<string> callersCacheNameList) {
            return create<NavigatorEntryModel>(core, recordGuid, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static NavigatorEntryModel createByName(CoreController core, string recordName) {
            return createByName<NavigatorEntryModel>(core, recordName);
        }
        //
        //====================================================================================================
        public static NavigatorEntryModel createByName(CoreController core, string recordName, ref List<string> callersCacheNameList) {
            return createByName<NavigatorEntryModel>(core, recordName, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public new void save(CoreController core) {
            base.save(core);
        }
        //
        //====================================================================================================
        public static void delete(CoreController core, int recordId) {
            delete<NavigatorEntryModel>(core, recordId);
        }
        //
        //====================================================================================================
        public static void delete(CoreController core, string ccGuid) {
            delete<NavigatorEntryModel>(core, ccGuid);
        }
        //
        //====================================================================================================
        public static List<NavigatorEntryModel> createList(CoreController core, string sqlCriteria, string sqlOrderBy, List<string> callersCacheNameList) {
            return createList<NavigatorEntryModel>(core, sqlCriteria, sqlOrderBy, callersCacheNameList);
        }
        //
        //====================================================================================================
        public static List<NavigatorEntryModel> createList(CoreController core, string sqlCriteria, string sqlOrderBy) {
            return createList<NavigatorEntryModel>(core, sqlCriteria, sqlOrderBy);
        }
        //
        //====================================================================================================
        public static List<NavigatorEntryModel> createList(CoreController core, string sqlCriteria) {
            return createList<NavigatorEntryModel>(core, sqlCriteria);
        }
        //
        //====================================================================================================
        public void invalidatePrimaryCache(CoreController core, int recordId) {
            invalidateCache<NavigatorEntryModel>(core, recordId);
        }
        //
        //====================================================================================================
        public static string getRecordName(CoreController core, int recordId) {
            return baseModel.getRecordName<NavigatorEntryModel>(core, recordId);
        }
        //
        //====================================================================================================
        public static string getRecordName(CoreController core, string ccGuid) {
            return baseModel.getRecordName<NavigatorEntryModel>(core, ccGuid);
        }
        //
        //====================================================================================================
        public static int getRecordId(CoreController core, string ccGuid) {
            return baseModel.getRecordId<NavigatorEntryModel>(core, ccGuid);
        }
        //
        //====================================================================================================
        public static NavigatorEntryModel createDefault(CoreController core) {
            return createDefault<NavigatorEntryModel>(core);
        }
    }
}
